src/main/java/jimmy/mcgymmy/logic/parser/McGymmyParser.java

```diff
@@ -50,14 +50,19 @@ public McGymmyParser(MacroList macroList) {
     public CommandExecutable parse(String text) throws ParseException {
         ParserUtil.HeadTailString headTail = ParserUtil.HeadTailString.splitString(text);
         String commandName = headTail.getHead();
-        if (commandName.equals("macro")) {
-            return parseCreateMacro(text);
-        } else if (this.macroList.hasMacro(commandName)) {
-            return this.parseRunMacro(commandName, headTail.getTail());
-        } else if (this.primitiveCommandParser.hasCommand(commandName)) {
-            return this.primitiveCommandParser.parsePrimitiveCommand(commandName, headTail.getTail());
-        } else {
-            throw new ParseException(Messages.MESSAGE_UNKNOWN_COMMAND);
+        try {
+            if (commandName.equals("macro")) {
+                return parseCreateMacro(text);
+            } else if (this.macroList.hasMacro(commandName)) {
+                return this.parseRunMacro(commandName, headTail.getTail());
+            } else if (this.primitiveCommandParser.hasCommand(commandName)) {
+                return this.primitiveCommandParser.parsePrimitiveCommand(commandName, headTail.getTail());
+            } else {
+                throw new ParseException(Messages.MESSAGE_UNKNOWN_COMMAND);
+            }
+        } catch (IllegalArgumentException e) {
```

Comment:
TODO will have to address this next time: each parsing function should catch this and rethrow as `ParseError`